osbs/api.py

```diff
@@ -394,9 +393,25 @@ def _create_build_config_and_build(self, build_request):
                                                        scheduled=True)
             logger.debug('Changed parent ImageStreamTag? %s', changed_ist)
 
+        old_nodeselector = existing_bc['spec'].get('nodeSelector')
+        new_nodeselector = None
+        logger.debug("saving build config's nodeSelector %s", old_nodeselector)
+
+        if old_nodeselector:
+            patch_nodeselector = {'spec': {'nodeSelector': nodeselector}}
+            logger.debug("setting build config's nodeSelector %s", nodeselector)
+            new_nodeselector = nodeselector
+            self.os.patch_build_config(build_config_name, json.dumps(patch_nodeselector))
+
+        elif nodeselector:
+            patch_nodeselector = {'spec': {'nodeSelector': nodeselector}}
+            logger.debug("setting build config's nodeSelector %s", nodeselector)
+            new_nodeselector = nodeselector
+            self.os.patch_build_config(build_config_name, json.dumps(patch_nodeselector))
+
         if triggers:
-            existing_bc['spec']['triggers'] = triggers
-            self.os.update_build_config(build_config_name, json.dumps(existing_bc))
+            patch_triggers = {'spec': {'triggers': triggers}}
+            self.os.patch_build_config(build_config_name, json.dumps(patch_triggers))
 
         if image_stream and triggers:
             prev_version = existing_bc['status']['lastVersion']
```

Comment:
Robert is right here, my comment is not detailed enough.

Patching would change `resourceVersion` in metadata - the version of the buildconfig. `lastVersion` here is the number of the last build. But since patching changes the buildconfig it may trigger the build (I'm not sure about it though), and thus changing `lastVersion`